backend/src/test/java/com/bakdata/conquery/integration/sql/testcontainer/hana/HanaContainer.java

```diff
@@ -0,0 +1,91 @@
+package com.bakdata.conquery.integration.sql.testcontainer.hana;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.nio.file.attribute.PosixFilePermission;
+import java.time.Duration;
+import java.util.Set;
+
+import lombok.SneakyThrows;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+public class HanaContainer<SELF extends HanaContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
+
+	public static final Integer DEFAULT_TENANT_HANA_PORT = 39041;
+	private static final String MASTER_PASSWORD = "HXEHana1";
+	private static final String USERNAME = "SYSTEM";
+	private static final String DATABASE_NAME = "HXE";
+
+	public HanaContainer(DockerImageName dockerImageName) {
+		super(dockerImageName);
+		prepareTmpHanaDir();
```

Comment:
ist IO im constructor hier wirklich sinnvoll? Gibt es dafür nicht eine Phase in der Container Lifetime?

Reply:
Das war ein guter Hinweis. Hab es jetzt umgebaut so dass das Erzeugen/Löschen des temporären Ordners im `@BeforeAll` und `@AfterAll` Hook nach den Hana-Tests ausgeführt wird.